File: employee-backend/src/main/java/com/example/employee_backend/service/EmployeeService.java
package com.example.employee_backend.service;

import com.example.employee_backend.dto.EmployeeDto;

import java.util.List;

public interface EmployeeService {

    // CREATE
    EmployeeDto createEmployee(EmployeeDto employeeDto);

    // READ ALL

    List<EmployeeDto> getAllEmployees();

    // READ BY ID
    EmployeeDto getEmployeeById(Long id);

    // UPDATE
    EmployeeDto updateEmployee(Long id, EmployeeDto employeeDto);

    // DELETE
    void deleteEmployee(Long id);
}

File: employee-frontend/src/services/employeeService.js
import api from "./api";

// Get all employees
export const getEmployees = () => {
  return api.get("/api/employees");
};

// Add employee
export const addEmployee = (employee) => {
  return api.post("/api/employees", employee);
};

// Update employee
export const updateEmployee = (id, employee) => {
  return api.put(`/api/employees/${id}`, employee);
};

// Delete employee
export const deleteEmployee = (id) => {
  return api.delete(`/api/employees/${id}`);
};

